lib/ansible/modules/system/user.py

```diff
@@ -1001,9 +1001,13 @@ def create_homedir(self, path):
                     shutil.copytree(skeleton, path, symlinks=True)
                 except OSError as e:
                     self.module.exit_json(failed=True, msg="%s" % to_native(e))
-        else:
+            else:
+                try:
+                    os.makedirs(path)
+                except OSError as e:
+                    self.module.exit_json(failed=True, msg="%s" % to_native(e))
             try:
-                os.makedirs(path)
+                os.chmod(path, 0o700)
```

Comment:
I don't believe we should be hard coding permissions here. We should use the system default value from `/etc/login.defs` or add this as a module parameter the defaults to the system value.

Reply:
Good call. I'll check the value in `/etc/login.defs` and possible add a new parameter.

Reply:
In the end I decided not to add a new parameter, because it would just complicate things in situations when a user home is being created with `useradd`. Now it tries to read `UMASK` from `/etc/login.defs` and if it it doesn't find it, it just uses the default.